homeassistant/components/sonos/sensor.py

```diff
@@ -0,0 +1,130 @@
+"""Entity representing a Sonos Move battery level."""
+import datetime
+import logging
+from typing import TYPE_CHECKING, Any, Dict, Optional
+
+from pysonos.core import SoCo
+from pysonos.exceptions import SoCoException
+
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.icon import icon_for_battery_level
+
+if TYPE_CHECKING:  # Avoid recursive import loops
+    from .media_player import SonosEntity
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = datetime.timedelta(seconds=10)
+
+ATTR_BATTERY_LEVEL = "battery_level"
+ATTR_BATTERY_CHARGING = "charging"
+ATTR_BATTERY_POWERSOURCE = "power_source"
+
+
+class SonosBatteryEntity(Entity):
+    """Representation of a Sonos Battery entity.
+
+    Defers connection status to parent SonosEntity.
+    """
+
+    def __init__(self, parent: "SonosEntity", soco: SoCo, battery_info: Dict[str, Any]):
+        """Initialize a SonosBatteryEntity from a parent SonosEntity."""
+        self._parent = parent
+        self._soco = soco
+        self._battery_info = battery_info
+        self._timer = None
+
+    @staticmethod
+    def fetch(soco: SoCo) -> Optional[Dict[str, Any]]:
+        """Fetch battery_info from the given SoCo object.
+
+        Returns None if the device doesn't support battery info
+        or if the device is offline.
+        """
+        try:
+            return soco.get_battery_info()
+        except (ConnectionError, TimeoutError, SoCoException):
+            pass
+        return None
+
+    async def async_added_to_hass(self) -> None:
+        """Register polling callback when added to hass."""
+        if self._timer is None:
+            self._timer = self._parent.hass.helpers.event.async_track_time_interval(
+                self.update, SCAN_INTERVAL
+            )
```

Comment:
timer will always be none. Instead of saving the timer, wrap the call to `async_track_time_interval` in `self.async_on_remove`